Docs: alertmanager datasource (#34567)
grafana/grafana: docs/sources/datasources/alertmanager.md

+++
title = "Alertmanager"
description = "Guide for using Alertmanager in Grafana"
keywords = ["grafana", "prometheus", "guide"]
aliases = ["/docs/grafana/latest/features/datasources/alertmanager"]
weight = 1300
+++

# Alertmanager data source

> Grafana includes built-in support for Prometheus Alertmanager. Add it as a datasource and you will be able to use the [Grafana alerting UI](https://grafana.com/docs/grafana/latest/alerting/) to manage silences, contact points and notification policies. A dropdown will appear in these pages to allow you to switch between Grafana and any configured alertmanager data sources This data source cannot return graph data and is not available in panels.

Reviewer: ```suggestion
Grafana includes built-in support for Prometheus Alertmanager. Once you add it as a data source, you can use the [Grafana unified alerting UI](https://grafana.com/docs/grafana/latest/alerting/) to manage silences, contact points as well as notification policies. A drop down option in these pages allows you to switch between Grafana and any configured Alertmanager data sources 

The Alertmanager data source cannot return graph data and is not available in panels.
```

Author: Are we publically calling this "unified" alerting? I thought "unified" is an internal designation

Author: updated, but as per slack discussion, w/o "unified" word

Refer to [Add a data source]({{< relref "add-a-data-source.md" >}}) for instructions on how to add a data source to Grafana. Only users with the organization admin role can add data sources.

> >**Note:** New in Grafana 8.0.

Reviewer: Remove note, as these docs are for 8.0.

Reviewer 2: How big a deal is this? Should we mention it in What's new in Grafana 8.0? @pkolyvas, thoughts?

Reviewer 2: Generally if we are adding a whole new topic to document something, I want it called out in the What's new. However, we have previously not done this (added docs) for alpha functionality.

Reviewer: This Alertmanager helps with the new alerting. So, a mention within that topic in What's New is probably the best.

Author: This is part of the new alerting system, not really a stand-alone thing. I don't see an entry for new alerting at all in what's new for 8.0, and I don't think it makes sense to tackle it in the scope of this PR. Is that OK?

Reviewer: > This is part of the new alerting system, not really a stand-alone thing. I don't see an entry for new alerting at all in what's new for 8.0, and I don't think it makes sense to tackle it in the scope of this PR. Is that OK?

Yes

Reviewer 2: Ah, gotcha. @achatterjee-grafana, please make sure it is linked in the What's new doc when the alerting feature entry is updated.

Reviewer: Will add this as a new data source.

> >**Note:** Currently only [Cortex Alertmanager](https://cortexmetrics.io/docs/proposals/scalable-alertmanager/) is supported.

Reviewer: Is this Cortex Alertmanager or Prometheus?

Author: both in a way, as in Cortex implementation of Prometheus Alertmanager :sweat_smile:

Reviewer: Can we make this little clear, then we should be good to go. Thanks!

Author: Updated to "Cortex implementation of Prometheus Alertmanager"

> >**Note:** Currently only [Cortex implementation of Prometheus alertmanager](https://cortexmetrics.io/docs/proposals/scalable-alertmanager/) is supported. 

Reviewer: ```suggestion
>**Note:** Currently only the [Cortex implementation of Prometheus alertmanager](https://cortexmetrics.io/docs/proposals/scalable-alertmanager/) is supported. 
```

> >**Note:** It is presently in alpha and not accessible unless alpha plugins are [enabled in Grafana settings](https://grafana.com/docs/grafana/latest/administration/configuration/#enable_alpha).

Reviewer: ```suggestion
It is presently in alpha and not accessible unless [alpha plugins are enabled in Grafana settings](https://grafana.com/docs/grafana/latest/administration/configuration/#enable_alpha).
```

> It is presently in alpha and not accessible unless [alpha plugins are enabled in Grafana settings](https://grafana.com/docs/grafana/latest/administration/configuration/#enable_alpha).

Reviewer: This should be the second line in the first paragraph.

Author: done!

## Provision the Alertmanager data source

> You can configure data sources using config files with Grafana's provisioning system. Read more about how it works and all the settings you can set for data sources on the [provisioning docs page]({{< relref "../administration/provisioning/#datasources" >}}).

Reviewer: ```suggestion
You can configure the Alertmanager data sources using config files with Grafana's provisioning system. For more information on how it works and the settings available, refer to the [provisioning docs page]({{< relref "../administration/provisioning/#datasources" >}}).
```

> You can configure the Alertmanager data sources using config files with Grafana's provisioning system. For more information on how it works and the settings available, refer to the [provisioning docs page]({{< relref "../administration/provisioning/#datasources" >}}).

Reviewer: ```suggestion
Configure the Alertmanager data sources by updating Grafana's configuration files. For more information on how it works and the settings available, refer to the [provisioning docs page]({{< relref "../administration/provisioning/#datasources" >}}).
```

> Here is an example for provisioning this datasource:

Reviewer: ```suggestion
Here is an example for provisioning the Alertmanager data source:
```
